api/data_explorer/swagger/swagger.yaml

```diff
@@ -17,8 +17,32 @@ paths:
       parameters: []
       responses:
         200:
-          description: "Facets."
+          description: "Success"
           schema:
-            type: "string"
+            $ref: "#/definitions/FacetsResponse"
       x-swagger-router-controller: "data_explorer.controllers.facets_controller"
-definitions: {}
+definitions:
+  FacetsResponse:
+    type: "array"
+    description: "Results from a faceted search."
+    items:
+      $ref: "#/definitions/Facet"
+  Facet:
```

Comment:
On the topic of naming, since the concept of a Facet will be pretty integral across our code base here, I'd like to be convinced that this is the best choice before we invest too heavily to change this. Facet fits syntactically with faceted search which is nice, are there any other reasons why we chose this naming? If the goal of the data explorer is solely to provide faceted search on datasets I think this is a good name, otherwise we might opt for something more widely recognized.

Alternatively, I like the name 'Field' as this is an instantly recognizable concept to anybody who has worked with tables and encapsulates the same idea. We could also use 'Feature' which is more familiar to ML but also conveys the same idea. Thoughts?

Reply:
I'm thinking about what api calls we need to the support. I'm pretty sure we'll need a text box that will autocomplete on field name.

A dataset may have hundreds of fields. 10 of those fields might be facets.

The UI calls /facets to specifically request the facet fields.

To support autocomplete, there could be a /fields call that searches over all hundreds of fields.

I think this should be called Facet, because the UI specifically wants facets to display in a facet UI. (And I think the React component should also be Facet.)